spacewalk/certs-tools/mgr_ssl_cert_setup.py

```diff
@@ -319,8 +319,8 @@ def getCertData(cert):
         elif line.startswith("    "):
             if nextval == "subjectKeyIdentifier":
                 data["subjectKeyIdentifier"] = line.strip().upper()
-            elif nextval == "authorityKeyIdentifier" and line.startswith("    keyid:"):
-                data["authorityKeyIdentifier"] = line[10:].strip().upper()
+            elif nextval == "authorityKeyIdentifier":
+                data["authorityKeyIdentifier"] = line.strip().upper()
```

Comment:
But this mean, that it will fail if you do not have openssl3, right?
Would it make sense to add a dependency to the spec file?

Reply:
Oh, this is getting more complicated :)
I have a company cert signed by a public CA, and the openssl output does not contain the 'keyid' string:
```
uyuni-server:/ # openssl x509 -noout -text -in ~/ssl/cert.crt | grep "Authority Key Identifier" -A 1 
            X509v3 Authority Key Identifier: 
                69:02:D7:4F:EC:56:3F:D0:0C:1B:EB:8A:1D:6B:39:91:B8:C0:0F:52
```
But when I try to parse the SUMA's own self-signed cert, there is a keyid:
```
uyuni-server:/ # openssl x509 -noout -text -in /srv/www/htdocs/pub/RHN-ORG-TRUSTED-SSL-CERT | grep "Authority Key Identifier" -A 3
            X509v3 Authority Key Identifier: 
                keyid:D7:B7:F6:C8:48:B7:2C:43:DB:E9:03:B6:D8:BC:F8:94:45:D2:4D:E7
                DirName:....
                serial:05:8C:CD:80:6F:2E:91:CD:D3:18:07:3F:BE:43:5E:01:E8:C1:13:64
```
You can also test with the suse.com cert:
```
uyuni-server:/ # openssl s_client -showcerts -connect suse.com:443 </dev/null 2>/dev/null| openssl  x509 -noout -subject -subject_hash -startdate -enddate -issuer -issuer_hash -modulus -ext subjectKeyIdentifier,authorityKeyIdentifier 
subject=CN = *.suse.com
4309e692
notBefore=Sep 17 00:00:00 2023 GMT
notAfter=Oct 15 23:59:59 2024 GMT
issuer=C = US, O = Amazon, CN = Amazon RSA 2048 M03
0264ec71
Modulus=936BB5ABF50AD69FACB7C6CC5019FDBD69131BAFECA48D1248A104A97D41B2007BCBA6ECDED0E5C624DE4308B7BE79213FB27E81B8C0BECE7D7D4355DF458191F28546CDB12E44C297F9D829FDD50AA69A52CE8675CF33CB00AA96122AFF1E7EEE000A08048F69A626B49A8366862382B739CA0C7D2AAAB2224E300696E9948BA3AD3DB608963F028A236772108C477861A25D4D89BDC2EE8F643EF1896EE90AE022856A96E56CCC0F7AF619B42131985D2D1712A4C6C5E351195A0DF7ED6FFF17C3DFA25783B0F34795D9C46597AFE824CA47D23C88B9A01E0F867694C15EF110DD2AE38D0078F75ADE614DC3E02AE89CA24AF3D9986F55EE44FBFF46CDAB8D
X509v3 Authority Key Identifier: 
    55:D9:18:5F:D2:1C:CC:01:E1:58:B4:BE:AB:D9:55:42:01:D7:2E:02
X509v3 Subject Key Identifier: 
    04:40:04:18:B2:86:76:E8:B2:A4:92:DE:BB:D5:95:B9:C3:17:9D:18
```